fe/src/main/java/org/apache/doris/analysis/CaseExpr.java

```diff
@@ -191,7 +191,7 @@ public void analyzeImpl(Analyzer analyzer) throws AnalysisException {
                 throw new AnalysisException("Subquery in case-when must return scala type");
             }
             if (whenExpr.contains(Predicates.instanceOf(Subquery.class))
-                    && !((hasCaseExpr() && whenExpr instanceof Subquery || whenExpr instanceof BinaryPredicate))) {
+                    && !((hasCaseExpr() && whenExpr instanceof Subquery || !checkSubquery(whenExpr)))) {
                 throw new AnalysisException("Only support subquery in binary predicate in case statement.");
```

Comment:
you need to modify exception message for you delete 'BinaryPredicate'

Reply:
Instead of deleting BinaryPredicate, we still only support BinaryPredicate, but just adjust the check of BinaryPredicate from the first layer to support check of children of other expr